### PR TITLE
Refactor runtime export checks. NFC

### PR DESCRIPTION
**Pull request by sbc100**
This saves a lot of codesize (albeit only in debug builds) and also
makes output JS easier to read IMHO.

### CODE REVIEW

**Comment by sbc100** [comment]
15% saving on size of debug JS for hello world!

File: src/modules.js [hunk 1]
```diff
@@ -368,16 +368,11 @@ function exportRuntime() {
     // if it is used, that they should export it
     if (ASSERTIONS) {
       // check if it already exists, to support EXPORT_ALL and other cases
-      // (we could optimize this, but in ASSERTIONS mode code size doesn't
-      // matter anyhow)
```
Comment (kripken) [inline src/modules.js:372]: PRs like this are good but I do think they are low priority, as the comment here said, `in ASSERTIONS mode code size doesn't matter anyhow`. Just wanted to check we're on the same page there?

**Review by kripken — APPROVED** [review]

File: src/modules.js (hunk 1 above)
Comment (sbc100) [reply src/modules.js:372]: Sure, this change wasn't really motivated by the code size saving originally... mostly it makes the output code easier to read.. at least to me.   Its also code that our user will look ad and I think that the excessive repetition is bad smell.    I sometimes try to read the emscripten output and this stuff always look wrong to me... but maybe its just me.

Comment (sbc100) [reply src/modules.js:372]: BTW the reason its so bad is that this code is output for all library symbols as well as runtime symbols .. so its a huge number!  That might be bug..?

Comment (kripken) [reply src/modules.js:372]: Ah, yes, if it makes the code more readable that's definitely worthwhile!

Hmm, that might be a bug... it does look wrong now that I look at it. It should only include things actually linked in, ideally.

**Pull request merged**